tests/test_serialize.py

```diff
@@ -0,0 +1,89 @@
+import io
+
+import pytest
+
+import py_adapter
+
+
+def tsst_invalid_format(ship_obj):
+    with pytest.raises(..., match="'does not exist' serialization format not supported"):
+        py_adapter.serialize(ship_obj, "does not exist")
+
+
+def test_serialize_json(ship_obj, ship_class):
+    data = py_adapter.serialize(ship_obj, format="JSON")
```

Comment:
How are we thinking about defining allowed formats here? An Enum? Just a string with some checks? I see you also have a comment about potentially auto-detecting

Reply:
We want this to be pluggable. So people can install packages which support different formats. Therefore, by definition, it will be a free-style text field. But obviously the error message could say which formats are supported in case of no match. 

Do we want case insensitive matching?
Do we want to expect a string which is also a valid Python variable name?
Do we suggest plugin developers namespace their formats to avoid clashes. E.g. a builtin-plugin "JSON" could easily clash with a more specific/opinionated JSON plugin. Although some of that could be handle using some sensible ordering (built-in comes last).

Reply:
auto-detecting is something we would need to think about and may not be worth it.